### DISCUSSION TITLE
Different PHP for site and PHPMyAdmin

### DISCUSSION THREAD

**Discussion by L-Gabor** [leokhoa/laragon#693] [Q&A]
I’ve kept Laragon and current sites I’m working on up to date in terms of PHP (8.1), MySQL (8.0) and Apache (2.4.58), but occasionally I need to run archived sites that worked on, say, PHP 5.5.  When I switch to PHP 5.5, however, I can’t run PHPMyAdmin any more because it needs at least PHP 7.2.5.  Is there any way to run PHPMyAdmin on a recent version while running my site on a legacy one?

**Comment by Pen-y-Fan** [comment]
There are a couple of options:

1. You can use Heidi SQL instead.
2. Laragon is portable, you can copy the laragon folder and have one with PHP 8.1 and the other with PHP 5.5 (with a compatible version of PHPMyAdmin)

My preference is option 1. Heidi SQL doesn't take long to learn.

**Reply by Pen-y-Fan** [reply]
Correct HeidiSQL is an executable app and independent of PHP.

**Comment by josealgardataborges** [comment]
> I’ve kept Laragon and current sites I’m working on up to date in terms of PHP (8.1), MySQL (8.0) and Apache (2.4.58), but occasionally I need to run archived sites that worked on, say, PHP 5.5. When I switch to PHP 5.5, however, I can’t run PHPMyAdmin any more because it needs at least PHP 7.2.5. Is there any way to run PHPMyAdmin on a recent version while running my site on a legacy one?

You can:

- UNZIP PHPMyAdmin version you want use in PHP V5 to C:\Laragon\www\phpmyadminV5
- UNZIP PHPMyAdmin version you want use in PHP V8 to C:\Laragon\www\phpmyadminV8

And when you need to use PHPMyAdmin (PHP5) change php version to 5, and access http://phpmyadminV5.test 

Or change PHP version to 8 and access http://phpmyadminV8.test 

(where .test is your local Virtual HOST)
![image](https://github.com/leokhoa/laragon/assets/8929655/6c3b07df-db26-4f48-934d-3a06bd3b23df)

You can also do, as suggested by @Pen-y-Fan, two separate LARAGON folders (Since it's portable). For instance:

Folder "**C:\LaragonLegacy**" for legacy PHP5 projects, where you have PHP5 and PHPMyAdmin setup.
Folder "**C:\Laragon**" with the current versions of PHP or PHPMyAdmin for day to day usage.

*BUT* and this is in my opinion the *BEST* solution, you should use a EXTERNAL BINARY to access Databases, there are a lot of free options, to name a few:

https://tableplus.com/ (Free, but you can only have 3 tabs open)

https://dev.mysql.com/downloads/workbench/ (from MySQL themselves but sometimes buggy)

https://dbgate.org/ (Free)

https://www.heidisql.com/ (Free)

https://www.beekeeperstudio.io/ (free)


--------------------------------------------------------------------------------------------------------------------------------------------

Choose whatever suits better your needs, definitely the best option is the last one and that way you will only need to use one *LARAGON* folder (like I do).

I use a lot of old PHP projects, so I have lots of PHP versions available to use:

![image](https://github.com/leokhoa/laragon/assets/8929655/3d0a4eea-9eb8-482c-ae43-b7a0bd4e90bd)

![image](https://github.com/leokhoa/laragon/assets/8929655/e95ee332-989e-43ac-a5fa-2707c1760db1)

![image](https://github.com/leokhoa/laragon/assets/8929655/a1961136-18fb-4538-8757-9a9f8dcc1bf9)

*ALL running a single LARAGON from one single folder*

**Comment by L-Gabor** [comment]
Thanks @Pen-y-Fan and @josealgardataborges for the helpful advice.  I ended up copping out and reverting to a backup I had of my entire laragon folder from when, for some reason, the sites I needed to worked with PHP 5.5, and when I stopped laragon and switched to PHP 8.2 I could do what I needed to in PHPMyAdmin.  Not sure why this switching stopped working for me, all I did was update Apache from 2.4.51 to 2.4.58, MySQL from 5.7.42 to 8.0.35, and just added versions of PHP.  Anyway, now I’m going to carefully do the updates again making sure I can still switch as needed.  I looked at Heidi SQL, but I’m so used to PHPMyAdmin I want to stay with that.  Will report back once all my updates are complete with any observations/issues.

**Reply by PrinzessinHana** [reply]
Is there any solution? I'm having the same problem with Drupal and phpMyAdmin when using PHP 8.5.5.
